src/NetEvolve.HealthChecks.Mongodb/MongoDbHealthCheck.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace NetEvolve.HealthChecks.MongoDb;
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using NetEvolve.Extensions.Tasks;
+using NetEvolve.HealthChecks.Abstractions;
+
+internal sealed class MongoDbHealthCheck : ConfigurableHealthCheckBase<MongoDbOptions>
+{
+    public MongoDbHealthCheck(IOptionsMonitor<MongoDbOptions> optionsMonitor)
+        : base(optionsMonitor) { }
+
+    protected override async ValueTask<HealthCheckResult> ExecuteHealthCheckAsync(
+        string name,
+        HealthStatus failureStatus,
+        MongoDbOptions options,
+        CancellationToken cancellationToken
+    )
+    {
+        using var client = new MongoClient(options.ConnectionString); // Ensure IDisposable is properly handled
```

Comment:
We should consider, that the Mongo client is already registered in the IServiceProvider. Like other Health checks. Also consider a keyed registration